chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -34,6 +36,28 @@ pub(crate) const LIMIT_PENDING_PEERS: usize = 60;
 /// We send these messages multiple times to reduce the chance that they are lost
 const IMPORTANT_MESSAGE_RESENT_COUNT: usize = 3;
 
+impl TryFrom<&PeerInfo> for WhitelistNode {
+    type Error = anyhow::Error;
+    fn try_from(pi: &PeerInfo) -> anyhow::Result<Self> {
+        Ok(Self {
+            id: pi.id.clone(),
+            addr: if let Some(addr) = pi.addr {
+                addr.clone()
+            } else {
+                anyhow::bail!("addess is missing");
+            },
+            account_id: pi.account_id.clone(),
+        })
+    }
+}
```

Comment:
drive by, but `TryFrom<&T>` like these look pretty ugly. TryFrom/From is usually about ownership transport or format changes, it's not a generall one-argument constructor. So, something like

```
impl WhitelistNode {
  pub fn from_peer_info(pi: &PeerInfo) -> anyhow::Result<Self>
}
```

would read better. 

The most immediate benefit is that its far easier to find all call-sites.

Reply:
done